content/en/docs/v3.5/learning/persistent-storage-files.md

```diff
@@ -0,0 +1,619 @@
+---
+title: etcd persistent storage files
+weight: 2650
+description: Reference of the persistent storage format and files
+---
+
+This document explains the etcd persistent storage format: naming, content and tools that allow developers to inspect them. Going forward the document should be extended with changes to the storage model. This document is targeted at etcd developers to help with their data recovery needs.
+
+
+## Prerequisites
+
+The following articles provide helpful background information for this document:
+
+* etcd data model overview: https://etcd.io/docs/v3.5/learning/data_model
+* Raft overview: http://web.stanford.edu/~ouster/cgi-bin/papers/Raft-atc14.pdf (especially "5.3 Log replication" section).
+
+
+## Overview
+
+### Long leaving files
+
+<table>
+  <tr>
+    <th>File name</th>
+    <th>High level purpose</th>
+  </tr>
+
+  <tr>
+    <td><pre>./member/snap/db</pre></td>
+    <td><strong>bbolt <a href="https://en.wikipedia.org/wiki/B%2B_tree">b+tree</a></strong> that stores all the applied data, membership authorization information & metadata. It’s aware of what's the last applied WAL log index (<a href="https://github.com/etcd-io/etcd/blob/a1ff0d5373335665b3e5f4cb22a538ac63757cb6/server/etcdserver/cindex/cindex.go#L92">"consistent_index"</a>).
```

Comment:
You have linked L92 here and also relevant lines other places, but with the development, these LOC number may change and then it won't point to an intended line and ead to incorrect pointers. So I would suggest to link to the specific class with a particular function name for reference. Thanks @ivanvc

Reply:
Hi @spzala, because the link points to a specific commit, the line number should always point to that line when the commit is done. Is it better to add the interface/struct and the function when referring to them? i.e., `func (*consistentIndex) UnsafeSave(backend.BatchTx): "consistent_index"` as the linked text rather than  `"consitent_index"`?

Reply:
Thanks @ivanvc Yes, that's better. With lines, what I am afraid is that as more codes/docs are added to those files, the lines won't correctly pointed from the links in the doc.

Reply:
Hey @spzala - I might be missing something but the links go to specific commits, which are immutable. The contents of those commits would never change.

Reply:
Ok, sounds good, thanks! Never mind, please @ivanvc @jmhbnz